src/llama-graph.cpp

```diff
@@ -1267,6 +1269,11 @@ ggml_tensor * llm_graph_context::build_attn_mha(
 
         ggml_tensor * kqv = ggml_mul_mat(ctx0, v, kq);
 
+        // for MLA with the absorption optimization, we need to "decompress" from MQA back to MHA
+        if (v_mla) {
+            kqv = ggml_mul_mat(ctx0, v_mla, kqv);
+        }
+
```

Comment:
This decompression can be performed after the `build_attn_mha`. This way, the FA path will not be coupled to MLA and we can extend it in the future.

I have prepare a fix in https://github.com/ggml-org/llama.cpp/commit/6dfbed009bd9e8e17bca97bb908e2f312a3166b7, so for now let's merge it like this and I will resolve this afterwards.